src/System.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -82,7 +82,19 @@ public SymmetricSignatureProvider(SymmetricSecurityKey key, string algorithm)
 
         public override bool IsSupportedAlgorithm(string algorithm)
         {
-            return false;
+            if (string.IsNullOrWhiteSpace(algorithm))
+                return false;
+
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.HmacSha1Signature:
```

Comment:
I'll update my PR by the end of the week. I'll also take a look at SHA-384 and SHA-512.